contrib/cmd/runkperf/commands/bench/read_update.go

```diff
@@ -0,0 +1,190 @@
+`,
+	Flags: append(
+		[]cli.Flag{
+			cli.IntFlag{
+				Name:  "total",
+				Usage: "Total requests per runner (There are 10 runners totally and runner's rate is 10)",
+				Value: 3600,
+			},
+			cli.StringFlag{
+				Name:  "read-update-namespace",
+				Usage: "Kubernetes namespace to use. If not specified, it will use the default namespace.",
+				Value: "default",
+			},
+			cli.IntFlag{
+				Name:  "read-update-configmap-total",
+				Usage: "Total ConfigMaps need to create",
+				Value: 100,
+			},
+			cli.IntFlag{
+				Name:  "read-update-configmap-size",
+				Usage: "Size of each ConfigMap. ConfigMap must not exceed 3 MiB.",
+				Value: 1024, // 1 MiB
+			},
+			cli.StringFlag{
+				Name:  "read-update-name-pattern",
+				Usage: "Name pattern for the resources to create",
+				Value: "kperf-read-update",
+			},
+			cli.Float64Flag{
+				Name:  "read-ratio",
+				Usage: "Proportion of read requests among all requests (range: 0.0 to 1.0). For example, 0.5 indicates 50% of the requests are reads.",
+				Value: 0.5,
+			},
+		},
+		commonFlags...,
+	),
+	Action: func(cliCtx *cli.Context) error {
+		_, err := renderBenchmarkReportInterceptor(
+			addAPIServerCoresInfoInterceptor(benchReadUpdateRun),
+		)(cliCtx)
+		return err
+	},
+}
+
+// benchReadUpdateRun is for subcommand benchReadUpdateRun.
+func benchReadUpdateRun(cliCtx *cli.Context) (*internaltypes.BenchmarkReport, error) {
+	ctx := context.Background()
+	kubeCfgPath := cliCtx.GlobalString("kubeconfig")
+
+	// Load the load profile
+	rgCfgFile, rgSpec, rgCfgFileDone, err := newLoadProfileFromEmbed(cliCtx,
+		"loadprofile/read_update.yaml")
+
+	if err != nil {
+		return nil, err
+	}
+	defer func() { _ = rgCfgFileDone() }()
+
+	total := cliCtx.Int("read-update-configmap-total")
+	size := cliCtx.Int("read-update-configmap-size")
+	namespace := cliCtx.String("read-update-namespace")
+	namePattern := cliCtx.String("read-update-name-pattern")
+	if total <= 0 || size <= 0 || total*size > 2*1024*1024 || size > 1024 {
+		return nil, fmt.Errorf("invalid total (%d) or size (%d) for configmaps: total must be > 0, size must be > 0, and total*size must not exceed 2 GiB, size must not exceed 1 MiB", total, size)
+	}
+
+	// Create configmaps with specified name pattern
+	client, err := utils.BuildClientset(kubeCfgPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build clientset: %w", err)
+	}
+
+	err = utils.CreateConfigmaps(ctx, kubeCfgPath, namespace, namePattern, total, size, 2, 0)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to create configmaps: %w", err)
+	}
+
+	defer func() {
+		// Delete the configmaps after the benchmark
+		err = utils.DeleteConfigmaps(ctx, kubeCfgPath, namespace, namePattern, 0)
+		if err != nil {
+			klog.Errorf("Failed to delete configmaps: %v", err)
+		}
+	}()
+
+	// Stop all the watches when the function returns
+	watches := make([]watch.Interface, 0)
+	defer stopWatches(watches)
```

Comment:
The underlying data referenced by watches may change when a scale-up occurs.
Using defer stopWatches(watches) directly will capture the old underlying data.

To ensure the evaluation is delayed until the function exits, you can wrap it as follows:

```go
defer func() {
       stopWatches(watches)
}()
```